base/partr.jl

```diff
@@ -19,9 +19,32 @@ const heap_d = UInt32(8)
 const heaps = [Vector{taskheap}(undef, 0), Vector{taskheap}(undef, 0)]
 const heaps_lock = [SpinLock(), SpinLock()]
 
+cong(max::UInt32) = iszero(max) ? UInt32(0) : jl_rand_ptls(max) + UInt32(1)
+
+function jl_rand_ptls(max::UInt32)
+    rngseed = Base.unsafe_load(Base.unsafe_convert(Ptr{UInt64}, Core.getptls()), 2) # TODO, less horrid
```

Comment:
@vtjnash any ideas on how to do this nicer?

Reply:
I made it so we only unsafe_convert plts once. Is that nice enough?